test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -37,6 +37,80 @@
    :ssl-key     localhost-key
    :ssl-ca-cert ca-cert})
 
+(def script-to-check-if-constant-is-defined
+  "begin; InstanceID; true; rescue NameError; false; end")
```

Comment:
Assuming you go with a "mock" approach for the `max-requests-flush-while-pool-flush-in-progress-test`, I'm thinking the WP violation might go away automatically - assuming you have to come up with an alternative way to the const for storing state like `set-constants-and-verify` does - one that hopefully wouldn't come with a warning message as a side effect.  I think this was the only test affected by the WP violation.